src/utils/getFilteredTodos.tsx

```diff
@@ -0,0 +1,23 @@
+import { FilterStatus } from '../types/FilterStatus';
+import { Todo } from '../types/Todo';
+
+export const getFilteredTodos = (todos: Todo[], filterStatus: FilterStatus) => {
+  let visibleTodos = [...todos];
+
+  if (filterStatus !== FilterStatus.ALL) {
+    visibleTodos = visibleTodos.filter(todo => {
+      switch (filterStatus) {
+        case FilterStatus.ACTIVE:
+          return !todo.completed;
+
+        case FilterStatus.COMPLETED:
+          return todo.completed;
+
+        default:
+          return true;
+      }
+    });
+  }
+
+  return visibleTodos;
```

Comment:
```suggestion
  if (filterStatus !== FilterStatus.ALL) {
    todos.filter(todo => {
      switch (filterStatus) {
        case FilterStatus.ACTIVE:
          return !todo.completed;

        case FilterStatus.COMPLETED:
          return todo.completed;

        default:
          return true;
      }
    });
  }

  return todos;
```
`filter` doesn't mutate an array, so you don't need to create a copy

Reply:
> `filter` doesn't mutate an array, so you don't need to create a copy

You forgot about this comments, fix it in the task